applications/app/controllers/CampaignsController.scala

```diff
@@ -0,0 +1,47 @@
+package controllers
+
+import common._
+import conf.Configuration
+import model._
+import play.api.libs.json._
+import play.api.libs.ws.{WSClient, WSResponse}
+import play.api.mvc._
+
+import scala.concurrent._
+
+class CampaignsController(
+  val controllerComponents: ControllerComponents,
+  val wsClient: WSClient,
```

Comment:
There is no need to prefix these parameters with `val`.
Actually, having this prefix makes them public fields whereas leaving it out makes them private constructor parameters.

Reply:
It throws up the error message: 
`class CampaignsController must be either declared abstract or implement abstract member 'controllerComponents' in 'play.api.mvc.BaseControllerHelpers'`